packages/@lwc/synthetic-shadow/src/faux-shadow/events.ts

```diff
@@ -63,6 +63,11 @@ interface ListenerMap {
     [key: string]: WrappedListener[];
 }
 
+const elementToListenerAndEventsMap: WeakMap<
```

Comment:
The `Set<string>` has to be cleaned up when an event listener is manually removed. I am suspecting that currently you can't add, remove and add again the same event listener on a given event target.

Reply:
yeah, nice catch